crates/common/tedge_config/src/tedge_config_cli/models/c8y.rs

```diff
@@ -0,0 +1,73 @@
+use std::str::FromStr;
+use strum::Display;
+
+/// A flag that switches legacy or advanced software management API.
+/// Can be set to auto in the future, see #2778.
+#[derive(
+    Debug, Clone, Copy, serde::Serialize, serde::Deserialize, Eq, PartialEq, doku::Document, Display,
+)]
+#[strum(serialize_all = "camelCase")]
+pub enum SoftwareManagementApiFlag {
+    Legacy,
+    Advanced,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to parse flag: {input}. Supported values are: legacy, advanced")]
+pub struct InvalidSoftwareManagementApiFlag {
+    input: String,
+}
+
+impl FromStr for SoftwareManagementApiFlag {
+    type Err = InvalidSoftwareManagementApiFlag;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        match input {
+            "legacy" => Ok(SoftwareManagementApiFlag::Legacy),
+            "advanced" => Ok(SoftwareManagementApiFlag::Advanced),
+            _ => Err(InvalidSoftwareManagementApiFlag {
+                input: input.to_string(),
+            }),
+        }
+    }
+}
+
+/// A flag that switches using the SmartREST static template 501-503 or 504-506 to update operation status.
+/// Can be set to auto in the future, see #2616.
+#[derive(
+    Debug, Clone, Copy, serde::Serialize, serde::Deserialize, Eq, PartialEq, doku::Document, Display,
+)]
+#[strum(serialize_all = "camelCase")]
+pub enum SmartRestWithOperationId {
+    Yes,
+    No,
+}
```

Comment:
I would not introduce a new type here and simply use `bool`.

- To be consistent with the other settings (e.g  `tedge config c8y.enable.device_profile true`)
- To be backward compatible with the [`Auto` type](https://github.com/thin-edge/thin-edge.io/blob/main/crates/common/tedge_config/src/tedge_config_cli/models/auto.rs#L8), in case we add support for `auto` in the future.

Reply:
Done in [ad57206](https://github.com/thin-edge/thin-edge.io/pull/3076/commits/ad57206d021cfbde1005a326544727697d781896)